x-pack/plugins/actions/server/routes/find.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import Hapi from 'hapi';
+
+import { WithoutQueryAndParams } from '../types';
+
+interface FindRequest extends WithoutQueryAndParams<Hapi.Request> {
+  query: {
+    per_page: number;
+    page: number;
+    search?: string;
+    default_search_operator: 'AND' | 'OR';
+    search_fields?: string[];
+    sort_field?: string;
+    has_reference?: {
+      type: string;
+      id: string;
+    };
+    fields?: string[];
+  };
+}
+
+export function findRoute(server: Hapi.Server) {
+  server.route({
+    method: 'GET',
+    path: `/api/action/_find`,
+    options: {
+      validate: {
+        query: Joi.object()
+          .keys({
+            per_page: Joi.number()
+              .min(0)
+              .default(20),
+            page: Joi.number()
+              .min(0)
+              .default(1),
```

Comment:
That's a good catch, I believe it would return an out of bound error on the `from` parameter when calling elasticsearch. I will fix that.

This is copied from saved objects API so I will make sure it gets fixed there as well.